### ISSUE TITLE
[KSP2] KSAnnotation.arguments sometimes triggers an NPE

### ISSUE THREAD

**Issue by BoD** [google/ksp#2112]
Upgrading our project to Kotlin `2.1.0-Beta1`, KSP `2.1.0-Beta1-1.0.25` and `ksp.useKSP2=true`, I am getting this:

```
Caused by: java.lang.NullPointerException: null cannot be cast to non-null type org.jetbrains.kotlin.load.java.structure.impl.JavaClassImpl
	at com.google.devtools.ksp.impl.symbol.kotlin.UtilKt.getDefaultValue(util.kt:635)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl$defaultArguments$2.invoke(KSAnnotationImpl.kt:110)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl$defaultArguments$2.invoke(KSAnnotationImpl.kt:84)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl.getDefaultArguments(KSAnnotationImpl.kt:84)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl$arguments$2.invoke(KSAnnotationImpl.kt:77)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl$arguments$2.invoke(KSAnnotationImpl.kt:72)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.google.devtools.ksp.impl.symbol.kotlin.KSAnnotationImpl.getArguments(KSAnnotationImpl.kt:72)
	at com.apollographql.execution.processor.ApolloProcessorKt.getArgumentValue(ApolloProcessor.kt:160)
```

Repro steps:
```
git clone git@github.com:apollographql/apollo-kotlin.git
cd apollo-kotlin
git checkout kotlin-2.1.0
git checkout 5cbb969d9dc04ea137abeb26bd3f1d35b6e12211
./gradlew --stacktrace :apollo-debug-server:build
```

The same stacktrace has been referenced in #885 but it's not clear if it is really the same issue or not.

**Comment by ting-yuan** [comment]
The fix will be available in the next, 1.0.26 release. If you couldn't wait, it is currently available in the [nightly build](https://github.com/google/ksp?tab=readme-ov-file#nightly-builds).

**Comment by BoD** [comment]
Thanks for the prompt reply @ting-yuan 🙏. Although, the nightly builds only contain builds for the stable version of Kotlin right? So no version for `2.1.0-Beta1` there.
